trieste/models/gpflow/sampler.py

```diff
@@ -240,16 +250,8 @@ def __init__(
         self._num_features = num_features  # m
         self._num_data = len(self._dataset.query_points)  # n
 
```

Comment:
perhaps we should add a runtime check that the model is instance of SupportsGetKernelObservationNoise, same as in acquisition functions?